expression/builtin_time_vec.go

```diff
@@ -1116,11 +1116,56 @@ func (b *builtinToDaysSig) vecEvalInt(input *chunk.Chunk, result *chunk.Column)
 }
 
 func (b *builtinDateFormatSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinDateFormatSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	buf, err := b.bufAllocator.get(types.ETDatetime, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf)
+	if err := b.args[0].VecEvalTime(b.ctx, input, buf); err != nil {
+		if err := handleInvalidTimeError(b.ctx, err); err != nil {
+			return err
+		}
+	}
+
+	buf0, err := b.bufAllocator.get(types.ETString, n)
+	if err != nil {
+		return err
+	}
+	defer b.bufAllocator.put(buf0)
+	if err := b.args[1].VecEvalTime(b.ctx, input, buf0); err != nil {
```

Comment:
This line should be ```if err := b.args[1].VecEvalString(b.ctx, input, buf0); err != nil {```

Reply:
Thank's so much for your help! I'm so careless.